rule.go

```diff
@@ -130,6 +130,12 @@ func (cc *Conn) AddRule(r *Rule) *Rule {
 		Data: append(extraHeader(uint8(r.Table.Family), 0), msgData...),
 	})
 
+	if cc.entities == nil {
+		cc.entities = make(map[int]Entity)
+	}
+
+	cc.entities[len(cc.messages)] = r
```

Comment:
Please help me understand `cc.entities` is a map.  How the number of messages can be a key  for a rule?

Reply:
I've no more idea on how to map netlink message before send (without netlink sequence) and netlink message from return of send. I tried to sort cc.entities as cc.messages, so when i add a rule, it's appened a the end so the len give me the index. I agree it's not perfect.

Reply:
I guess it is one of the reason why @stapelberg left it to a higher level implementation. Some time ago I reached out to nftables/netfilter community with the same question and they had strongly advised me to use `UserData` field for this purpose. Maybe you could use the same approach as I did, it works well.